File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int beg = 0, end = numbers.length - 1;
        while (beg < end) {
            int total = numbers[beg] + numbers[end];
            if (total == target) return new int[] { beg + 1, end + 1 }; // 1-based indices
            if (total > target) end--;
            else beg++;
        }
        return new int[] { -1, -1 }; // if no pair found
    }
}

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left, right = 0, len(numbers) - 1

        while left < right:
            tot =numbers[left] + numbers[right]
            if tot == target:
                return [left +1, right + 1]
            if tot > target:
                right -= 1
            if tot < target:
                left += 1

